### PR TITLE
Swift 5 Migration

### PR DESCRIPTION
**Pull request by nick-potts**
I redid the migration as per #594 , it also resolves #611 . Additionally, it added quite a few redundant roundtrip helper functions so I removed them.

### CODE REVIEW

**Comment by nick-potts** [comment]
CI is down :/

**Comment by JoeMatt** [comment]
Yes, GitHub changed their default macOS image for their docker environment. Mac is actually outsourced to Orka, and for some reason a couple of weeks ago they changed what their default SDK installs would be.

for open source projects, it's not a huge deal, you have to now install any but the default iOS simulator versions for each XCode version, which takes a minute or two. I hit this snag in my commercial repos where the time hit kind of sucks now that we gotta make a value cost for the extra minutes for every push.

I'll try to push a fix soon, but this PR would then have to merge or rebase that change.

https://www.macstadium.com/orka

image

File: Podfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 target 'HeroExamples' do
   platform :ios, '9.0'
   use_frameworks!
-  pod 'CollectionKit'
+  pod 'CollectionKit', :inhibit_warnings => true
```
Comment (kuyazee) [inline Podfile]: Shouldn't we inhibit warnings for all pods?

**Comment by JoeMatt** [comment]
Since your own pod is a pod for the demo app than no. Though you would inhibit all and then use false for just hero. Same difference

Joe Mattiello
________________________________

**Comment by kuyazee** [comment]
CI fixed in #648 let's rerun the CI after it gets merged

**Comment by nick-potts** [comment]
I rebased to get CI to run but its broken again haha

**Comment by JoeMatt** [comment]
Taking a look into the CI issue.

**Comment by JoeMatt** [comment]
This was rebased, conflict resolved and merged in pr #696

**Pull request closed without merging**